remote_config/src/android/remote_config_android.cc

```diff
@@ -1073,7 +1201,55 @@ ConfigUpdateListenerRegistration*
 RemoteConfigInternal::AddOnConfigUpdateListener(
     std::function<void(ConfigUpdate&&, RemoteConfigError)>
         config_update_listener) {
-  return nullptr;
+  JNIEnv* env = app_.GetJNIEnv();
+  // Create a struct to wrap the listener function.
+  ConfigUpdateListenerWrapper* listener_wrapper =
+      new ConfigUpdateListenerWrapper(config_update_listener);
+  // Create a Java Listener and give it a pointer to the listener struct.
+  jobject j_listener =
+      env->NewObject(jni_config_update_listener::GetClass(),
+                     jni_config_update_listener::GetMethodId(
+                         jni_config_update_listener::kConstructor),
+                     reinterpret_cast<jlong>(listener_wrapper));
+  FIREBASE_ASSERT(!util::CheckAndClearJniExceptions(env));
+
+  jobject j_local_registration = env->CallObjectMethod(
+      internal_obj_, config::GetMethodId(config::kAddOnConfigUpdateListener),
+      j_listener);
+  FIREBASE_ASSERT(!util::CheckAndClearJniExceptions(env));
+
+  // Create a global reference to the registration
+  jobject j_registration = env->NewGlobalRef(j_local_registration);
```

Comment:
If we need a global reference, we would need to define that in the class level, and delete it in app destroy phase.

Reply:
Or you can delete it at the end of this function. but that kinda dismiss the purpose of the global ref

Reply:
i see the pattern here, yeah you would need the global ref inside the callback, but you would need to delete it inside as well.

Reply:
Ok. I delete the global ref in the removal function. I also added a bool to make sure the removal function is not called twice so that the ref is not deleted twice.